cpp/src/interop/from_arrow_device.cu

```diff
@@ -0,0 +1,448 @@
+      skip_mask ? nullptr : reinterpret_cast<bitmask_type const*>(input->buffers[0]);
+    auto data_buffer = input->buffers[1];
+    return std::make_tuple<column_view, owned_columns_t>(
+      {type, num_rows, data_buffer, null_mask, static_cast<size_type>(input->null_count), offset},
+      {});
+  }
+};
+
+column_view get_empty_type_column(size_type size)
+{
+  return {data_type(type_id::EMPTY), size, nullptr, nullptr, size};
+}
+
+std::tuple<column_view, owned_columns_t> get_column(ArrowSchemaView* schema,
+                                                    const ArrowArray* input,
+                                                    data_type type,
+                                                    bool skip_mask,
+                                                    rmm::cuda_stream_view stream,
+                                                    rmm::mr::device_memory_resource* mr);
+
+template <>
+std::tuple<column_view, owned_columns_t> dispatch_to_cudf_column::operator()<numeric::decimal128>(
+  ArrowSchemaView* schema,
+  const ArrowArray* input,
+  data_type type,
+  bool skip_mask,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
+{
+  size_type const num_rows = input->length;
+  size_type const offset   = input->offset;
+  bitmask_type const* null_mask =
+    skip_mask ? nullptr : reinterpret_cast<bitmask_type const*>(input->buffers[0]);
+  auto data_buffer = input->buffers[1];
+  return std::make_tuple<column_view, owned_columns_t>(
+    {type, num_rows, data_buffer, null_mask, static_cast<size_type>(input->null_count), offset},
+    {});
+}
+
+template <>
+std::tuple<column_view, owned_columns_t> dispatch_to_cudf_column::operator()<bool>(
+  ArrowSchemaView* schema,
+  const ArrowArray* input,
+  data_type type,
+  bool skip_mask,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
+{
+  if (input->length == 0) {
+    return std::make_tuple<column_view, owned_columns_t>(
+      {type,
+       0,
+       nullptr,
+       skip_mask ? nullptr : reinterpret_cast<bitmask_type const*>(input->buffers[0]),
+       0},
+      {});
+  }
+  auto out_col         = mask_to_bools(reinterpret_cast<bitmask_type const*>(input->buffers[1]),
+                               input->offset,
+                               input->offset + input->length,
+                               stream,
+                               mr);
+  auto const has_nulls = skip_mask ? false : input->buffers[0] != nullptr;
+  if (has_nulls) {
+    auto out_mask =
+      cudf::detail::copy_bitmask(reinterpret_cast<bitmask_type const*>(input->buffers[0]),
+                                 input->offset,
+                                 input->offset + input->length,
+                                 stream,
+                                 mr);
+    out_col->set_null_mask(std::move(out_mask), input->null_count);
+  }
+
+  auto out_view = out_col->view();
+  owned_columns_t owned;
+  owned.emplace_back(std::move(out_col));
```

Comment:
I get the same ` error: static assertion failed with "result type must be constructible from value type of input range"` error that I get when trying to do brace initialization if I try to do this. That's why I set it up using the `emplace_back` like this.

Reply:
Ah I see, this is probably happening because the initializer list is const too so that further makes it impossible for the compiler to deduce any constructor.